topic-operator/src/main/java/io/strimzi/operator/topic/v2/ReplicasChangeClient.java

```diff
@@ -0,0 +1,373 @@
+import static io.strimzi.operator.topic.v2.TopicOperatorUtil.hasReplicasChange;
+import static io.strimzi.operator.topic.v2.TopicOperatorUtil.topicNames;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.stream.Collectors.groupingBy;
+
+/**
+ * Replicas change client for Cruise Control API.
+ * <br/><br/>
+ * The REST endpoints are {@code topic_configuration} to request replication factor changes and {@code user_tasks} 
+ * to check the asynchronous execution result. Cruise Control runs one task execution at a time, additional requests 
+ * are queued up to {@code max.active.user.tasks}.
+ * <br/><br/>
+ * At any given time, a KafkaTopic.spec.replicas change can be in one of the following states:
+ * <br/>
+ * <ul><li>Pending: Not in Cruise Control's task queue (not yet sent or request error).</li>
+ * <li>Ongoing: In Cruise Control's task queue, but execution not started, or not completed.</li>
+ * <li>Completed: Cruise Control's task execution completed (target replication factor reconciled).</li></ul>
+ */
+public class ReplicasChangeClient {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(ReplicasChangeClient.class);
+    private static final String USER_TASK_ID_HEADER = "User-Task-ID";
+    private static final long REQUEST_TIMEOUT_SEC = 60;
+    
+    private final TopicOperatorConfig config;
+    private final ExecutorService httpClientExecutor;
+    private final ObjectMapper mapper;
+    
+    /**
+     * Create a new replicas change client instance.
+     * 
+     * @param config Topic Operator configuration.
+     */
+    public ReplicasChangeClient(TopicOperatorConfig config) {
+        this.config = config;
+        this.httpClientExecutor = Executors.newCachedThreadPool();
+        this.mapper = new ObjectMapper();
+    }
+
+    /**
+     * Send a topic_configuration request to create a task for replication factor change of one or more topics.
+     * This should be called when one or more .spec.replicas changes are detected.
+     * 
+     * @param reconcilableTopics Pending replicas changes.
+     * @return Replicas changes with status update.
+     */
+    public List<ReconcilableTopic> requestPendingChanges(List<ReconcilableTopic> reconcilableTopics) {
+        List<ReconcilableTopic> result = new ArrayList<>();
+        if (reconcilableTopics.isEmpty()) {
+            return result;
+        }
+        updateToPending(reconcilableTopics, "Replicas change pending");
+        result.addAll(reconcilableTopics);
+        
+        try {
+            LOGGER.debugOp("Sending topic configuration request, topics {}", topicNames(reconcilableTopics));
+            HttpClient client = buildHttpClient();
+            client.sendAsync(buildTopicConfigPostRequest(reconcilableTopics), HttpResponse.BodyHandlers.ofString())
+                .thenAccept(response -> {
+                    if (response.statusCode() != 200) {
```

Comment:
Just out of curiosity...If there is any 2xx code would we treat that as a fail? I am not sure if there is such a scenario but it might be worth considering at least

Reply:
This is not documented, but looking at CC code only javax.servlet.http.HttpServletResponse.SC_OK (200) is returned for success.